### PR TITLE
JS GitHub workflows

### PR DESCRIPTION
**Pull request by SenseException**

### CODE REVIEW

**Comment by greg0ire** [comment]
Should [`paths`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore) also be specified?

**Comment by SenseException** [comment]
That would look like in the recent commit.

File: .github/workflows/js-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+on:
+  push:
+    paths:
+      - '**.js'
+      - '**.scss'
+      - 'package.json'
+      - 'yarn.lock'
+      - 'jest.config.js'
+      - 'webpack.config.js'
+      - '.github/workflows/js-tests.yml'
```
Comment (greg0ire) [inline .github/workflows/js-analysis.yml:12]: :thinking: why would we need to run that other workflow when changing this one and vice versa?

File: .github/workflows/js-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+on:
+  push:
+    paths:
+      - '**.js'
+      - '**.scss'
+      - 'package.json'
+      - 'yarn.lock'
+      - 'jest.config.js'
+      - 'webpack.config.js'
```
Comment (greg0ire) [inline .github/workflows/js-analysis.yml:11]: Are these 2 lines not redundant with the first one?

**Review by derrabus — APPROVED** [review]

File: .github/workflows/js-analysis.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      - 'yarn.lock'
+      - '.github/workflows/js-analysis.yml'
+
+jobs:
+  code-analysis:
+
+    runs-on: "ubuntu-latest"
+
+    steps:
+    - uses: "actions/checkout@v2"
```
Comment (greg0ire) [inline .github/workflows/js-analysis.yml:25]:
```suggestion
    - uses: "actions/checkout@v3"
```

File: .github/workflows/js-tests.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      - 'yarn.lock'
+      - '.github/workflows/js-tests.yml'
+
+jobs:
+  js-tests:
+
+    runs-on: "ubuntu-latest"
+
+    steps:
+    - uses: "actions/checkout@v2"
```
Comment (greg0ire) [inline .github/workflows/js-tests.yml:25]:
```suggestion
    - uses: "actions/checkout@v3"
```

**Review by greg0ire — APPROVED** [review]

**Comment by greg0ire** [comment]
Approving in case you used "v2" in order to be consistent with other workflows, and plan to switch all workflows to v3 in another PR.

**Comment by SenseException** [comment]
> Approving in case you used "v2" in order to be consistent with other workflows, and plan to switch all workflows to v3 in another PR.

Revisiting the workflows wasn't at the top of my todo list, but I can do at least update versions in a new PR in one of my next PRs.

**Pull request merged**